docs/pages/access-controls/reference.mdx

```diff
@@ -373,3 +374,188 @@ Here is an explanation of the fields used in the `where` and `filter` conditions
+available when defining roles.
+
+#### `{{internal.kube_username}}`
+
+- **Type**: Variable
+- **Description**: Kubernetes usernames available to the user.
+- **Example**: `{{internal.kube_username}}` -> `["myuser", "kube-admin"]`.
+
+#### `{{internal.kubernetes_groups}}`
+
+- **Type**: Variable
+- **Description**: List of allowed Kubernetes groups.
+- **Example**: `{{internal.kubernetes_groups}}` -> `["group1", "group2"]`
+
+#### `{{internal.kubernetes_users}}`
+
+- **Type**: Variable
+- **Description**: List of allowed Kubernetes users.
+- **Example**: `{{internal.kubernetes_users}}` -> `["user1", "user2"]`
+
+#### `{{internal.logins}}`
+
+- **Type**: Variable
+- **Description**: List of allowed SSH logins.
+- **Example**: `{{internal.logins}}` -> `["user1", "user2"]`
+
+#### `{{internal.windows_logins}}`
+
+- **Type**: Variable
+- **Description**: List of allowed Windows logins.
+- **Example**: `{{internal.windows_logins}}` -> `["user1", "user2"]`
+
+### External Variables
+
+These variables provide context from an Identity Provider or other external data source. We can't predict every variable an external data source will provide, but these are common values expected from SAML and OIDC identity providers.
+
+#### `{{external.logins}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.username}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.access["label"]}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.env}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.first_name}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's first name from the SSO provider.
+- **Example**: `{{external.first_name}}` -> `"Alice"`
+- **Provider**: OIDC/SAML
+
+**Note**: Per OIDC specifications this value may be given as `given_name`.
+
+#### `{{external.last_name}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's last name from the SSO provider.
+- **Example**: `{{external.last_name}}` -> `"Smith"`
+- **Provider**: OIDC/SAML
+
+**Note**: Per OIDC specifications this value may be given as `family_name`.
+
+#### `{{external.email}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's email address from the SSO provider.
+- **Example**: `{{external.email}}` -> `"alice@example.com"`
+- **Provider**: OIDC/SAML
+
+#### `{{external.groups}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's group membership from the SSO provider.
+- **Example**: `{{external.groups}}` -> `["group1", "group2"]`
+- **Provider**: OIDC/SAML
+
+#### `{{external.azure_identities}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's Azure identities from the SSO provider [(2)].
+- **Example**: `{{external.azure_identities}}` -> `["identity1"]`
+- **Provider**: OIDC/SAML
+
+#### `{{external.databases}}`
```

Comment:
I'm not sure about documenting all of these as separate variables, it's really just a claim the comes from your IdP, it could be anything. `groups` and `email` will probably be the most common by far, many of the others would have to be manually configured in most cases I think

Reply:
Is this concern addressed by the content at the start of the section?

> These variables provide context from an Identity Provider or other external data source. We can't predict every variable an external data source will provide, but these are common values expected from SAML and OIDC identity providers.

If not, what would you propose we change?

Reply:
I would propose that we don't document each of these individually, we would have some general info about the `external` trait syntax, and then maybe just a simple list of common trait names (i think email and groups are common, everything else they would have been a custom trait they intentionally set up).